Paco/src/com/pacoapp/paco/js/bridge/JavascriptEventLoader.java

```diff
@@ -56,18 +56,95 @@ public String loadAllEvents() {
     return convertExperimentResultsToJsonString;
   }
 
+  //TODO: check if method has references
   @JavascriptInterface
   public String getLastEvent() {
-    // TODO make this class manage retrieval better so that we aren't pulling tons of data into the webview.
-    List<Event> events = experimentProviderUtil.loadEventsForExperimentByServerId(experiment.getId());
-    return FeedbackActivity.convertLastEventToJsonString(events);
+	  return getLastNEvents("1");
+  }
+  
+  @JavascriptInterface
+  public String getLastNEvents(String numberOfRecords) {
+	// TODO: Should this be 10; Adding a default value of 10
+	int noOfRecords=10;
+	try{  
+		noOfRecords = Integer.parseInt(numberOfRecords);
+	}catch(NumberFormatException nfe){
+		Log.error("Not a valid number of records :" + numberOfRecords);
+	}
+	long t1 = System.currentTimeMillis();
+    List<Event> events = experimentProviderUtil.loadEventsForExperimentByServerId(experiment.getId(), noOfRecords);
+    long t2 = System.currentTimeMillis();
+    Log.info("Time for getLastNEvents: " + (t2 - t1));
+    return FeedbackActivity.convertEventsToJsonString(events);
   }
 
   @JavascriptInterface
   public String getEventsForExperimentGroup() {
     List<Event> events = experimentProviderUtil.loadEventsForExperimentGroup(androidExperiment.getId(), experimentGroup.getName());
     return FeedbackActivity.convertEventsToJsonString(events);
   }
+  
+  @JavascriptInterface
+  public String getEventsByQuery(String criteriaQuery) {
```

Comment:
Please add documentation (could be a copy of the JS one) for the criteriaQuery parameter